hal/halx86/legacy/bussupp.c

```diff
@@ -854,36 +854,58 @@ HalpDebugPciDumpBus(IN ULONG i,
     if (VendorName)
     {
         /* Copy the vendor name into our buffer */
-        VendorName += 8;
+        VendorName += strlen("\r\n0000  ");
         p = strpbrk(VendorName, "\r\n");
         Length = p - VendorName;
         if (Length >= sizeof(bVendorName)) Length = sizeof(bVendorName) - 1;
         strncpy(bVendorName, VendorName, Length);
         bVendorName[Length] = '\0';
+        p += strlen("\r\n");
+        while (*p == '\t' || *p == '#')
+        {
+            p = strpbrk(p, "\r\n");
+            p += strlen("\r\n");
+        }
+        Boundary = p;
 
         /* Isolate the product name */
         sprintf(LookupString, "\t%04x  ", PciData->DeviceID);
         ProductName = strstr(VendorName, LookupString);
+        if (Boundary && ProductName >= Boundary)
+        {
+            ProductName = NULL;
+        }
         if (ProductName)
         {
             /* Copy the product name into our buffer */
-            ProductName += 7;
+            ProductName += strlen("\t0000  ");
             p = strpbrk(ProductName, "\r\n");
             Length = p - ProductName;
             if (Length >= sizeof(bProductName)) Length = sizeof(bProductName) - 1;
             strncpy(bProductName, ProductName, Length);
             bProductName[Length] = '\0';
+            p += strlen("\r\n");
+            while ((*p == '\t' && *(p + 1) == '\t') || *p == '#')
+            {
+                p = strpbrk(p, "\r\n");
```

Comment:
Is it possible that strpbrk returns NULL in this condition?

Reply:
Same for the second loop.